static/js/stream_create.js

```diff
@@ -307,99 +330,71 @@ export function show_new_stream_modal() {
         $("#announce-new-stream").hide();
     }
     clear_error_display();
-
-    $("#stream-checkboxes label.checkbox").on("change", function (e) {
-        const elem = $(this);
-        const stream_id = Number.parseInt(elem.attr("data-stream-id"), 10);
-        const checked = elem.find("input").prop("checked");
-        const subscriber_ids = new Set(peer_data.get_subscribers(stream_id));
-
-        $("#user-checkboxes label.checkbox").each(function () {
-            const user_elem = $(this);
-            const user_id = Number.parseInt(user_elem.attr("data-user-id"), 10);
-
-            if (subscriber_ids.has(user_id)) {
-                user_elem.find("input").prop("checked", checked);
-            }
-        });
-
-        e.preventDefault();
-    });
 }
 
 export function create_handlers_for_users(container) {
     // container should be $('#people_to_add')...see caller to verify
-    container.on("change", "#user-checkboxes input", update_announce_stream_state);
-
-    // 'Check all' and 'Uncheck all' visible users
-    container.on("click", ".subs_set_all_users", (e) => {
-        $("#user-checkboxes .checkbox").each((idx, li) => {
-            if (li.style.display !== "none") {
-                $(li.firstElementChild).prop("checked", true);
+    function update_checked_state_for_users(value, users) {
+        // Only do full_redraw if we have to render more than 500 users.
+        const full_redraw = !users || users.length > 500;
+        for (const user of all_users) {
```

Comment:
I guess this is O(n) regardless of how many users are modified by the click; would it be possible to do this instead by looping through `users` to make it vary with the number of things changed?

(Probably not worth a lot of complexity to arrange that).

Reply:
Yeah if we wanted to do this, we'd need to rework things to use a dictionary as the core data structure for `all_users`.  Probably not worth it given that and the fact some clicks will be `O(n)` which is hopefully fine.